libkbfs/folder_update_prepper.go

```diff
@@ -572,7 +572,18 @@ func (fup folderUpdatePrepper) updateResolutionUsage(ctx context.Context,
 // addUnrefToFinalResOp makes a resolutionOp at the end of opsList if
 // one doesn't exist yet, and then adds the given pointer as an unref
 // block to it.
-func addUnrefToFinalResOp(ops opsList, ptr BlockPointer) opsList {
+func addUnrefToFinalResOp(ops opsList, ptr BlockPointer,
+	doNotUnref map[BlockPointer]bool) opsList {
+	// Make sure the block ID we want to unref isn't in the "do not
+	// unref" list -- it could mean that block has already been GC'd
+	// by the merged branch.  We can't compare pointers directly
+	// because GC'd pointers contain no block context.
+	for noUnref := range doNotUnref {
```

Comment:
Yeah, the key of `doNotUnref` is a `BlockPointer`, not a `kbfsblock.ID`, so that won't work.  And it needs to stay a `BlockPointer` for other reasons elsewhere in the code unfortunately.